Fix `RichTextLabel` table top padding

Chenyang341 opened the pull request:
Fixes #103228.
Now while adding top padding to a cell, it no longer covers the cell above and no longer leaves a gap between
itself and the cell bellow. This was due to the position which the cell would be drawn being calculated incorrectly when there was a value of top padding different from 0.

Before when enabling BBCode on a RichTextLabel and populating with:
```
[table=1]
[cell bg=black padding=9,9,9,0]Padding of 9, 9, 9, 0[/cell]
[cell bg=black padding=9,0,9,0]Padding of 9, 0, 9, 0[/cell]
[cell bg=black padding=9,0,9,0]Padding of 9, 0, 9, 0[/cell]
[cell bg=black padding=9,0,9,9]Padding of 9, 0, 9, 9[/cell]
[/table]
```
It would generate the table:
![table1](https://github.com/user-attachments/assets/7b1a3bc8-0352-4a6a-8058-f2e9015cf79d)
Now it generates the table:
![table2](https://github.com/user-attachments/assets/baac75de-338f-4b5f-bfd7-d8e1f92ba47a)


<!--
Please t

arget the `master` branch in priority.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->

bruvzg reviewed (changes requested):
This will break selection/meta links in the padded cells, `_find_click_in_line` should be changed as well:

```diff
diff --git a/scene/gui/rich_text_label.cpp b/scene/gui/rich_text_label.cpp
index cbb2f62263..022bbdbeea 100644
--- a/scene/gui/rich_text_label.cpp
+++ b/scene/gui/rich_text_label.cpp
@@ -1610,7 +1610,7 @@ float RichTextLabel::_find_click_in_line(ItemFrame *p_frame, int p_line, const V
                                                                        }
                                                                        if (crect.has_point(p_click)) {
                                                                                for (int j = 0; j < (int)frame->lines.size(); j++) {
-                                                                                       _find_click_in_line(frame, j, rect.position + Vector2(frame->padding.position.x, frame->lines[j].offset.y), rect.size.x, 0, p_click, &table_click_frame, &table_click_line, &table_click_item, &table_click_char, true, p_meta);
+                                                                                       _find_click_in_line(frame, j, rect.position + Vector2(frame->padding.position.x, frame->lines[j].offset.y + frame->padding.position.y), rect.size.x, 0, p_click, &table_click_frame, &table_click_line, &table_click_item, &table_click_char, true, p_meta);
                                                                                        if (table_click_frame && table_click_item) {
                                                                                                // Save cell detected cell hit data.
                                                                                                table_range = Vector2i(INT32_MAX, 0);
```

Chenyang341 committed:
Fix #103228: Rich text table top padding issue fixed.
Now while adding top padding to a cell, it no longer covers
the cell above and no longer leaves a gap between
itself and the cell bellow.

Chenyang341 commented:
> This will break selection/meta links in the padded cells, `_find_click_in_line` should be changed as well:
> 
> ```diff
> diff --git a/scene/gui/rich_text_label.cpp b/scene/gui/rich_text_label.cpp
> index cbb2f62263..022bbdbeea 100644
> --- a/scene/gui/rich_text_label.cpp
> +++ b/scene/gui/rich_text_label.cpp
> @@ -1610,7 +1610,7 @@ float RichTextLabel::_find_click_in_line(ItemFrame *p_frame, int p_line, const V
>                                                                         }
>                                                                         if (crect.has_point(p_click)) {
>                                                                                 for (int j = 0; j < (int)frame->lines.size(); j++) {
> -                                                                                       _find_click_in_line(frame, j, rect.position + Vector2(frame->padding.position.x, frame->lines[j].offset.y), rect.size.x, 0, p_click, &table_click_frame, &table_click_line, &table_click_item, &table_click_char, true, p_meta);
> +                                                                                       _find_click_in_line(frame, j, rect.position + Vector2(frame->padding.position.x, frame->lines[j].offset.y + frame->padding.position.y), rect.size.x, 0, p_click, &table_click_frame, &table_click_line, &table_click_item, &table_click_char, true, p_meta);
>                                                                                         if (table_click_frame && table_click_item) {
>                                                                                                 // Save cell detected cell hit data.
>                                                                                                 table_range = Vector2i(INT32_MAX, 0);
> ```

I have changed the requested part. Thank you for the feedback.

Chenyang341 commented:
I don't have commit access, could you please merge it for me? Thanks.

bruvzg commented:
Vertical padding was significantly changed in https://github.com/godotengine/godot/pull/104685, not sure if this change is still relevant. I'll check it later.

bruvzg commented:
With #104685 changes, padding seems to be working as expected. I'm not sure if line spacing should or shouldn't be included, but this is not directly related. So this change is no longer necessary.

| Without this change (current master) | With this change (applied on top of current master) |
| :----: | :----: |
| <img width="398" alt="Screenshot 2025-04-01 at 09 55 26" src="https://github.com/user-attachments/assets/641fe47f-cbf5-4d7f-9c5f-3efba74c8b4f" /> | <img width="398" alt="Screenshot 2025-04-01 at 18 56 55" src="https://github.com/user-attachments/assets/b1dac042-6c73-4191-8177-97d845df908e" /> |

Thanks for your contribution nonetheless.
